src/runtime/internal/dom.ts

```diff
@@ -332,6 +332,13 @@ export function set_custom_element_data(node, prop, value) {
 	}
 }
 
+export function set_dynamic_element_data(tag: string) {
+	return function (node: HTMLElement, data_map: { [x: string]: string }) {
+		if (/-/.test(tag)) set_custom_element_data_map(node, data_map);
+		else set_attributes(node, data_map);
+	};
```

Comment:
We can move the if block out. Less code an a little bit faster probably
```suggestion
	return (/-/.test(tag)) ? set_custom_element_data_map : set_attributes;
```